File: 2021_summer_algorithm_study_intermediate/using_cpp/1222.cpp
/*
input :
5
4 6 3 8 9

output :
9
*/

#include <iostream>
#include <algorithm>
using namespace std;
using ll = long long;

// 전역 변수 선언
int N, num;
ll participant_num;
int cnt[2000005];
int max_val = -1;
ll ans = -1;

int main(){
    ios::sync_with_stdio(false); cin.tie(nullptr); cout.tie(nullptr);
    freopen("input.txt", "r", stdin);

    cin >> N;
    while (N--){
        cin >> num;
        cnt[num]++;
        max_val = max(max_val, num);
    }

    for (ll i = 1; i <= max_val; i++){
        participant_num = 0;
        for (int j = i; j <= max_val; j += i){
            participant_num += cnt[j];
        }

        if (participant_num >= 2){
            ans = max(ans, participant_num * i);
        }
    }
    cout << ans;

    return 0;
}

File: 2021_summer_algorithm_study_intermediate/using_python/1222.py
"""
input :
5
4 6 3 8 9

output :
9
"""

import sys
sys.stdin = open("input.txt", "r")
input = sys.stdin.readline


if __name__ == "__main__":
    N = int(input())
    nums = list(map(int, input().rstrip().split()))

    loop_cnt = max(nums)
    cnt = [0 for _ in range(loop_cnt + 1)]
    for num in nums:
        cnt[num] += 1

    ans = -1
    # 아래와 같은 loop의 시간복잡도는 NlogN이다.
    for i in range(1, loop_cnt + 1):
        participant_num = 0
        for j in range(i, loop_cnt + 1, i):
            participant_num += cnt[j]

        if participant_num >= 2:
            ans = max(ans, participant_num * i)
    print(ans)
